File: 20180712/Two SumII.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        if (numbers == null || numbers.length < 2){
            return new int[]{} ;
        }
        int left = 0;
        int right = numbers.length - 1;
        while(left <= right){
            if(numbers[left] + numbers[right] < target){
                left ++;
            } else if (numbers[left] + numbers[right] > target){
                right --;
            } else {
                return new int[]{left + 1, right + 1};
            }
        }
        return new int[]{};
    }
}

File: 20180712/Two SumII.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i in range(len(numbers)):
            if( target - numbers[i] in dic):
                return [dic[target - numbers[i]] + 1, i + 1]
            else:
                dic[numbers[i]] = i
        